src/solvers/strings/array_pool.cpp

```diff
@@ -135,12 +139,19 @@ static void attempt_assign_length_from_type(
   std::unordered_map<array_string_exprt, exprt, irep_hash> &length_of_array,
   symbol_generatort &symbol_generator)
 {
-  DATA_INVARIANT(
-    array_expr.id() != ID_if,
-    "attempt_assign_length_from_type does not handle if exprts");
-  // This invariant seems always true, but I don't know why.
-  // If we find a case where this is violated, try calling
-  // attempt_assign_length_from_type on the true and false cases.
+  if(array_expr.id() == ID_if)
+  {
+    const auto &if_expr = to_if_expr(array_expr);
+    attempt_assign_length_from_type(
+      to_array_string_expr(if_expr.true_case()),
+      length_of_array,
+      symbol_generator);
+    attempt_assign_length_from_type(
+      to_array_string_expr(if_expr.false_case()),
+      length_of_array,
+      symbol_generator);
```

Comment:
that doesn't seem correct, the length could be assigned twice with different value, while it only has to be equal to one of these depending on the condition